# Week 10 Assignment: Document Classification

**Kevin Martin**  
**DATA 620**  
**Spring 2026**

## Project Goal
For this assignment, I will use a labeled dataset of spam and non-spam emails to build a document classification model. The goal is to train the model on existing labeled documents and then evaluate how well it predicts whether new documents are spam or ham.

In [2]:
import pandas as pd

# Load dataset (we will download it next)
url = "https://archive.ics.uci.edu/ml/machine-learning-databases/spambase/spambase.data"

df = pd.read_csv(url, header=None)

print(df.shape)
df.head()

(4601, 58)


,0,1,2,3,4,5,6,7,8,9,...,48,49,50,51,52,53,54,55,56,57
0,0.00,0.64,0.64,0.0,0.32,0.00,0.00,0.00,0.00,0.00,...,0.00,0.000,0.0,0.778,0.000,0.000,3.756,61,278,1
1,0.21,0.28,0.50,0.0,0.14,0.28,0.21,0.07,0.00,0.94,...,0.00,0.132,0.0,0.372,0.180,0.048,5.114,101,1028,1
2,0.06,0.00,0.71,0.0,1.23,0.19,0.19,0.12,0.64,0.25,...,0.01,0.143,0.0,0.276,0.184,0.010,9.821,485,2259,1
3,0.00,0.00,0.00,0.0,0.63,0.00,0.31,0.63,0.31,0.63,...,0.00,0.137,0.0,0.137,0.000,0.000,3.537,40,191,1
4,0.00,0.00,0.00,0.0,0.63,0.00,0.31,0.63,0.31,0.63,...,0.00,0.135,0.0,0.135,0.000,0.000,3.537,40,191,1


In [3]:
# Check basic info about the dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4601 entries, 0 to 4600
Data columns (total 58 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   0       4601 non-null   float64
 1   1       4601 non-null   float64
 2   2       4601 non-null   float64
 3   3       4601 non-null   float64
 4   4       4601 non-null   float64
 5   5       4601 non-null   float64
 6   6       4601 non-null   float64
 7   7       4601 non-null   float64
 8   8       4601 non-null   float64
 9   9       4601 non-null   float64
 10  10      4601 non-null   float64
 11  11      4601 non-null   float64
 12  12      4601 non-null   float64
 13  13      4601 non-null   float64
 14  14      4601 non-null   float64
 15  15      4601 non-null   float64
 16  16      4601 non-null   float64
 17  17      4601 non-null   float64
 18  18      4601 non-null   float64
 19  19      4601 non-null   float64
 20  20      4601 non-null   float64
 21  21      4601 non-null   float64
 22  

In [4]:
# Separate features and target
X = df.iloc[:, :-1]   # all columns except last
y = df.iloc[:, -1]    # last column

print(X.shape)
print(y.shape)

(4601, 57)
(4601,)


In [5]:
!pip install scikit-learn

In [6]:
from sklearn.model_selection import train_test_split

# Split the data (70% train, 30% test)
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.3, random_state=42
)

print(X_train.shape)
print(X_test.shape)

(3220, 57)
(1381, 57)


In [7]:
from sklearn.naive_bayes import GaussianNB

# Create the model
model = GaussianNB()

# Train the model
model.fit(X_train, y_train)

print("model trained")

model trained


In [8]:
from sklearn.metrics import accuracy_score

# Make predictions
y_pred = model.predict(X_test)

# Check accuracy
accuracy = accuracy_score(y_test, y_pred)
print("Accuracy:", accuracy)

Accuracy: 0.8247646632874729


In [9]:
from sklearn.metrics import confusion_matrix

# Confusion matrix
cm = confusion_matrix(y_test, y_pred)
print(cm)

[[592 212]
 [ 30 547]]


## Model Evaluation

The model achieved an accuracy of approximately 0.825, which indicates strong overall performance.

The confusion matrix shows that the model is effective at correctly identifying spam emails, with a high number of true positives. However, there are some false positives, where non-spam emails are incorrectly classified as spam.

This suggests that while the model is reliable for detecting spam, it may occasionally flag legitimate emails as spam. Overall, the model demonstrates a good balance but could be improved to reduce false positives.

## Conclusion

In this project, I built a document classification model using the Spambase dataset to classify emails as spam or not spam. I trained a Naive Bayes classifier and evaluated its performance using accuracy and a confusion matrix.

The model achieved an accuracy of approximately 82.5%, showing that it can effectively distinguish between spam and non-spam emails. The confusion matrix provided additional insight into the model’s strengths and weaknesses, particularly its ability to detect spam while occasionally misclassifying some legitimate emails.

Overall, this project demonstrated how machine learning can be applied to document classification tasks and reinforced the importance of evaluating model performance beyond just accuracy.